external-crates/move/crates/move-compiler/src/sui_mode/linters/random_objects.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This analysis flags uses of random::Random and random::RandomGenerator in public functions.
+
+use crate::{
+    diag,
+    diagnostics::codes::{custom, DiagnosticInfo, Severity},
+    expansion::ast::Visibility,
+    naming::ast as N,
+    shared::{program_info::TypingProgramInfo, CompilationEnv},
+    typing::{ast as T, visitor::TypingVisitor},
+};
+use move_ir_types::location::Loc;
+use move_symbol_pool::Symbol;
+
+use super::{
+    LinterDiagCategory, LINTER_DEFAULT_DIAG_CODE, LINT_WARNING_PREFIX,
+    RANDOM_GENERATOR_STRUCT_NAME, RANDOM_MOD_NAME, RANDOM_STRUCT_NAME, SUI_PKG_NAME,
+};
+
+const RANDOM_OBJECTS_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::RandomObjects as u8,
+    LINTER_DEFAULT_DIAG_CODE,
+    "Risky use of random::Random or random::RandomGenerator in a public function",
+);
+
+pub struct RandomObjectsVisitor;
+
+impl TypingVisitor for RandomObjectsVisitor {
+    fn visit(
+        &mut self,
+        env: &mut CompilationEnv,
+        _program_info: &TypingProgramInfo,
+        program: &mut T::Program_,
+    ) {
+        for (_, _, mdef) in program.modules.iter() {
+            if mdef.attributes.is_test_or_test_only() {
```

Comment:
I think we will need to add filtering for the sui package too? Otherwise the Random module itself is going to light up?